### PR TITLE
Ruby Tic-Tac-Toe Project: Milestone 3: Game Logic

### PR DESCRIPTION
**Pull request by Leandro-Barretoo**
After finishing milestone 2 we started working on the game logic:
We added the following:
-Add placement of the symbol on board;
-Create conditions to verify if it was a winning move or not;
-Determine the number of moves and if no successful turns then its a tie;
-Add a rematch functionality;

### CODE REVIEW

**Comment by Leandro-Barretoo** [comment]
We have done only one commit cause due to the waiting for the milestone 2 code review we were working on relpt.it.
After milestone 2 was approved, milestone 3 was already made so it's just one commit with all the work. sorry for the inconvenience.

File: bin/main.rb
Comment (cristianCeamatu) [inline bin/main.rb]: - [ ] [Optional] Allways use `require_relative` for internal dependencies (like files you create) and `require` for external dependencies (likes gems or packages, more on this in the Rails module) - [link with details](https://github.com/rubocop-hq/ruby-style-guide#use-require_relative-whenever-possible)

File: bin/main.rb
Comment (cristianCeamatu) [inline bin/main.rb]: - [ ] [Please fix]  You are not respecting the DRY rule here, as a developer this rule is very important and it means that you always have to look for ways to not repeat yourself, like creating methods that do exactly the same thing. You need to convert this method to accept a parameter (ex: player) and perform the action depending on the parameter.
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: lib/player.rb
Comment (cristianCeamatu) [inline lib/player.rb]: - [ ] [Please fix] You have a similar situation here, two methods that do exactly the same thing. You need to think about accepting a parameter and returning the result depending on that.

File: lib/board.rb
Comment (cristianCeamatu) [inline lib/board.rb]: - [ ] Remove this line or correct it because it is blocking the game to start, the path is broken resulting in an error. I tried to remove the line along with the `< Play` from the `Board` class definition and the game works perfectly.

**Review by cristianCeamatu — CHANGES_REQUESTED** [review]
## ♻️ Changes required
#### Congratulations for your efforts! The logic was easy to follow and understand and you made use of a very simple loop to give the possibility to rematch!
However, there are some small changes to be done in order to get this project approved:

### Please check my inline comments and do the required changes then submit for another review

[Optional]

1.  When somebody wins the game, provide the name of the winner in the message.
2. Move the `Game` class on its own file. (you need to move the methods that deal with user input or print something outside of the class and inside of the `main.rb` file.

[For future projects]
Don`t work on the repl website, Microverse recommendation is to create a new branch and work on the next milestone, this way you learn how to solve merge conflicts in case you encounter any, this is a vital skill that you need as a developer when collaborating on complex projects.

Most of the commits are done only by Leandro, the reason behind this is most probably because you worked on repl, please avoid this situation and try to keep a balance between switching roles.

Don't hesitate to contact me on Slack if needed.

Best regards,
Cristian Ceamatu
Slack: Cristian Viorel Ceamatu

**Comment by M-Alamgir** [comment]
Thanks @cristianCeamatu for your review,
We have made your suggested changes.

**Review by ebeagusamuel — CHANGES_REQUESTED** [review]
# Changes Required:recycle:
### Hey there :wave: 
#### You have done well so far but there are still a few fixes that you need to make. 
- [x] There is an error when I try to run the game. This error is occurring because you are using `require` to load files. Always use `require_relative` when loading a file from the same project directory.
- [x] You game doesn't handle a draw. Please update the game to be able to handle a draw.
#### Request another review when you are done making these changes.
slack(@ebeagusamuel)

**Review by LamiaSristy — APPROVED** [review]
# Status: Approved. 💯 🥇 

Congratulations, @Leandro-Barretoo and @ourandco, Your project is now approved 👏 👏 👏 👏 .

Nice job, implementing all the required changes by the previous reviewers. 🛠️ 

You can merge the branch now, good luck with your next milestone 🚀.

- [ Optional]: Consider printing the winning player's name with the message.

Feel free to reach me at @lamiasristy 🙋

**Pull request merged**
